remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/verification/AesEncryptor.kt

```diff
@@ -0,0 +1,101 @@
+package com.rakuten.tech.mobile.remoteconfig.verification
+
+import android.os.Build
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import android.util.Base64
+import javax.crypto.Cipher
+import javax.crypto.spec.GCMParameterSpec
+import androidx.annotation.RequiresApi
+import androidx.annotation.VisibleForTesting
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import java.security.KeyStore
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+
+@RequiresApi(Build.VERSION_CODES.M)
+internal class AesEncryptor @VisibleForTesting constructor(
+    private val keyStore: KeyStore,
+    private val keyGenerator: AesKeyGenerator
+) : Encryptor {
+
+    constructor() : this(
+        keyStore = KeyStore.getInstance("AndroidKeyStore"),
+        keyGenerator = AesKeyGenerator(alias = KEYSTORE_ALIAS, provider = "AndroidKeyStore")
+    )
+
+    private val encryptionKey get() = (
+        keyStore.getEntry(KEYSTORE_ALIAS, null) as KeyStore.SecretKeyEntry?
+    )?.secretKey ?: keyGenerator.generateKey()
+
+    init {
+        keyStore.load(null)
+    }
+
+    override fun encrypt(data: String): String {
+        val cipher = Cipher.getInstance(CIPHER_TRANSFORMATION)
+        cipher.init(Cipher.ENCRYPT_MODE, encryptionKey)
+        val encryptedData = cipher.doFinal(data.toByteArray(Charsets.UTF_8))
+
+        return AesEncryptedData(
+            Base64.encodeToString(cipher.iv, Base64.DEFAULT),
```

Comment:
It might be worth checking how random the default generated IV is. 
https://stackoverflow.com/a/31037133/2910267 
https://doridori.github.io/Android-Security-Beware-of-the-default-IV/

Reply:
Hmm, I didn't know about that. 

It looks like it should be okay here though because I'm using `KeyGenParameterSpec` which sets `indistinguishability under chosen-plaintext attack` to true by default: https://developer.android.com/reference/android/security/keystore/KeyGenParameterSpec.Builder#setRandomizedEncryptionRequired(boolean)

But it's still probably safe to set it to true manually.